service/service_construction.go

```diff
@@ -1237,6 +1508,21 @@ func isUnwrapRequest(metadata map[string]interface{}) bool {
 	return false
 }
 
+func isGenericContractCall(metadata map[string]interface{}) bool {
+	if isUnwrap, ok := metadata["bridge_unwrap"]; ok {
+		if isUnwrap.(bool) {
```

Comment:
Can you check the panic here?

Reply:
Do u mean checking `bridge_unwrap` not existing in the `metadata` map ? According to the `blockchain-avacchain` repo, this is a valid case as for native and token transfer, metadata map does not have this field.